### ISSUE TITLE
Bug after update

### ISSUE THREAD

**Issue by GoldenGnu** [peterhaneve/evemon#62]
I got the following after updating to 4.0.6.
Running EveMon again, and it's working without any problems.
Just wanted to report it, even though it wasn't a major problem.
Thank you very much for working on EveMon 👍 
```
EVEMon Version: 4.0.6.4838
.NET Runtime Version: 4.0.30319.42000
Operating System: Microsoft Windows NT 10.0.17134.0
Executable Path: "C:\Program Files (x86)\EVEMon\EVEMon.exe"
System.NullReferenceException: Objektreferencen er ikke indstillet til en forekomst af et objekt.
   ved EVEMon.Common.Models.Character.GetLastKnownLocationText() i EVEMon.Common\Models\Character.cs:linje 548
   ved EVEMon.CharacterMonitoring.CharacterMonitorHeader.UpdateInfoControls() i EVEMon\CharacterMonitoring\CharacterMonitorHeader.cs:linje 230
   ved EVEMon.CharacterMonitoring.CharacterMonitorHeader.UpdateInfrequentControls() i EVEMon\CharacterMonitoring\CharacterMonitorHeader.cs:linje 208
   ved System.Windows.Forms.Control.OnParentVisibleChanged(EventArgs e)
   ved System.Windows.Forms.Control.OnVisibleChanged(EventArgs e)
   ved System.Windows.Forms.Control.OnParentVisibleChanged(EventArgs e)
   ved System.Windows.Forms.Control.OnVisibleChanged(EventArgs e)
   ved System.Windows.Forms.Control.SetVisibleCore(Boolean value)
   ved System.Windows.Forms.TabPage.set_Visible(Boolean value)
   ved System.Windows.Forms.TabControl.UpdateTabSelection(Boolean updateFocus)
   ved System.Windows.Forms.TabControl.OnSelectedIndexChanged(EventArgs e)
   ved System.Windows.Forms.TabControl.WmSelChange()
   ved System.Windows.Forms.TabControl.WndProc(Message& m)
   ved System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


Datafile report:
  eve-blueprints-en-US.xml.gzip (201KiB - 219eb76e35e9c1df32d2b7f21268582c)
  eve-certificates-en-US.xml.gzip (45KiB - 540000f7c111354769a54fa31e57d411)
  eve-geography-en-US.xml.gzip (1009KiB - 9a7a4fc5ca3340dc0c91f3116810d107)
  eve-items-en-US.xml.gzip (2403KiB - 00ce5d7ad1efce9283e08b2c4249ecbe)
  eve-masteries-en-US.xml.gzip (21KiB - 3668b50996fc40a0c0853fec427b5712)
  eve-properties-en-US.xml.gzip (61KiB - 1f047494a5dbcd31d9675257dc49098a)
  eve-reprocessing-en-US.xml.gzip (110KiB - e66052e6ad68555d0e14479a2b352280)
  eve-skills-en-US.xml.gzip (24KiB - 8684157679e5345a8b91d5ce15e0dd4f)


Diagnostic Log:
0d 0h 00m 00s > Starting up
0d 0h 00m 00s > EveMonClient.Initialize - begin
0d 0h 00m 00s > Program.StartupAsync - done
0d 0h 00m 00s > Settings.TryDeserializeFromFile - begin
0d 0h 00m 06s > Settings.TryDeserializeFromFile - done
0d 0h 00m 06s > Settings.Import - begin
0d 0h 00m 06s > EveMonClient.OnSchedulerChanged
0d 0h 00m 06s > Settings.Import - done
0d 0h 00m 06s > EveMonClient.OnSettingsChanged
0d 0h 00m 06s > Main loop - start
0d 0h 00m 06s > MainWindow.OnLoad
0d 0h 00m 06s > TimeCheck.ScheduleCheck - in 00:00:01
0d 0h 00m 06s > UpdateManager.ScheduleCheck - in 00:00:10
0d 0h 00m 06s > Main window - loaded
0d 0h 00m 06s > Datafiles.Load - begin
0d 0h 00m 07s > TimeCheck.BeginCheckAsync
0d 0h 00m 07s > TimeCheck.OnCheckCompleted - Synchronised
0d 0h 00m 07s > TimeCheck.ScheduleCheck - in 1.00:00:00
0d 0h 00m 08s > EveMonClient.OnServerStatusUpdated
0d 0h 00m 09s > Datafiles.Load - done
0d 0h 00m 09s > Settings.ImportData - begin
0d 0h 00m 09s > EveMonClient.OnCharacterImplantSetCollectionChanged - Golden Gnu
0d 0h 00m 09s > EveMonClient.OnCharacterSkillQueueUpdated - Golden Gnu
0d 0h 00m 09s > EveMonClient.OnCharacterUpdated - Golden Gnu
0d 0h 00m 09s > EveMonClient.OnCharacterImplantSetCollectionChanged - GoldenGnu
0d 0h 00m 09s > EveMonClient.OnCharacterSkillQueueUpdated - GoldenGnu
0d 0h 00m 09s > EveMonClient.OnCharacterUpdated - GoldenGnu
0d 0h 00m 09s > EveMonClient.OnEveIDToNameUpdated
0d 0h 00m 09s > EveMonClient.OnCharacterInfoUpdated - Golden Gnu
0d 0h 00m 09s > EveMonClient.OnCharacterInfoUpdated - GoldenGnu
0d 0h 00m 09s > EveMonClient.OnCharacterInfoUpdated - Silver Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterImplantSetCollectionChanged - Silver Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterSkillQueueUpdated - Silver Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterUpdated - Silver Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterImplantSetCollectionChanged - Glazeg
0d 0h 00m 10s > EveMonClient.OnCharacterSkillQueueUpdated - Glazeg
0d 0h 00m 10s > EveMonClient.OnCharacterUpdated - Glazeg
0d 0h 00m 10s > EveMonClient.OnCharacterCollectionChanged
0d 0h 00m 10s > EveMonClient.OnESIKeyCollectionChanged
0d 0h 00m 10s > EveMonClient.OnCharacterPlanCollectionChanged - Golden Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterPlanCollectionChanged - GoldenGnu
0d 0h 00m 10s > EveMonClient.OnCharacterPlanCollectionChanged - Silver Gnu
0d 0h 00m 10s > EveMonClient.OnCharacterPlanCollectionChanged - Glazeg
0d 0h 00m 10s > EveMonClient.OnMonitoredCharactersChanged
0d 0h 00m 10s > EveMonClient.OnMonitoredCharactersChanged
0d 0h 00m 11s > EveMonClient.OnMonitoredCharactersChanged
0d 0h 00m 11s > Settings.ImportData - done
0d 0h 00m 11s > EveMonClient.OnSettingsChanged
0d 0h 00m 12s > EveMonClient.OnCharacterUpdated - Golden Gnu
0d 0h 00m 12s > EveMonClient.OnCharacterListUpdated - 131755955969496896 (Golden Gnu)
0d 0h 00m 12s > EveMonClient.OnESIKeyCollectionChanged
0d 0h 00m 12s > EveMonClient.OnESIKeyInfoUpdated - 131755955969496896 (Golden Gnu)
0d 0h 00m 12s > EveMonClient.OnCharacterUpdated - Silver Gnu
0d 0h 00m 12s > EveMonClient.OnCharacterListUpdated - 131755956319672637 (Silver Gnu)
0d 0h 00m 12s > EveMonClient.OnESIKeyCollectionChanged
0d 0h 00m 12s > EveMonClient.OnESIKeyInfoUpdated - 131755956319672637 (Silver Gnu)
0d 0h 00m 12s > EveMonClient.OnCharacterUpdated - GoldenGnu
0d 0h 00m 12s > EveMonClient.OnCharacterListUpdated - 131755956187669685 (GoldenGnu)
0d 0h 00m 12s > EveMonClient.OnESIKeyCollectionChanged
0d 0h 00m 12s > EveMonClient.OnESIKeyInfoUpdated - 131755956187669685 (GoldenGnu)
0d 0h 00m 13s > EveMonClient.OnCharacterResearchPointsUpdated - Golden Gnu
0d 0h 00m 13s > EveMonClient.OnCharacterEVEMailMessagesUpdated - Silver Gnu
0d 0h 00m 13s > EveMonClient.OnCharacterEVENotificationsUpdated - Silver Gnu
0d 0h 00m 13s > EveMonClient.OnEveIDToNameUpdated
0d 0h 00m 13s > EveMonClient.OnCharacterInfoUpdated - Glazeg
0d 0h 00m 16s > UpdateManager.OnCheckCompleted
0d 0h 00m 16s > UpdateManager.ScheduleCheck - in 12:00:00
0d 0h 00m 21s > CitadelStationProvider.FetchIDs - ESI lookup for 1021153079633 using 131755956187669685 (GoldenGnu)
```

**Comment by wbSD** [comment]
This can be reproduced on a character that is docked in a structure, by removing that structure from the cache located in `AppData\Roaming\EVEMon\cache\xml\ConquerableStationList.xml`

The first time that Character.LastKnownStation is accessed, it triggers a query for the structure and returns the "Inaccessible Structure" in the meantime.
That structure has a SolarSystemID of 0, which causes its SolarSystem property to become null, which is what causes this error.

Subsequent queries for that structure id will return the correct structure, as it has now been fetched and cached.

**Comment by peterhaneve** [comment]
This issue should be fixed by fb957ed.
